third_party/pcsc-lite/naclport/server/src/google_smart_card_pcsc_lite_server/global.h

```diff
@@ -38,6 +43,30 @@ namespace google_smart_card {
 // already been initialized.
 void InitializeAndRunPcscLiteServer();
 
+class PcscLiteServerGlobal final {
+ public:
+  explicit PcscLiteServerGlobal(pp::Instance* pp_instance);
+  ~PcscLiteServerGlobal();
+  void Detach();
+
+  static const PcscLiteServerGlobal* GetInstance();
+
+  void InitializeAndRunPcscLiteServer();
+
+  void PostReaderInitAddMessage(const char* reader_name, int port,
+                                const char* device) const;
+  void PostReaderFinishAddMessage(const char* reader_name, int port,
+                                  const char* device, long return_code) const;
+  void PostReaderRemoveMessage(const char* reader_name, int port) const;
+
+ private:
+  pp::Instance* GetPPInstance() const;
+  void PostMessage(const char* type, const pp::VarDictionary& message_data) const;
+
+  pp::Instance* pp_instance_;
+  mutable std::mutex mutex_;
```

Comment:
I'd recommend to place the mutex member before pp_instance, in order to give a hint that the pp_instance_ field is protected by the mutex.